app/api.ts

```diff
@@ -202,14 +203,11 @@ export interface DialogData extends EntityData {
   messages: Array<Link<EncryptedTaggedByteView<Array<MessageData|ServiceMessageData>>>>
 }
 
-export interface DialogInfo {
-  id?: ToString<EntityID>
-  title: string
+export interface DialogInfo extends EntityData {
   initials: string
-  thumbSrc: string
   isPublic: boolean
-  type: EntityType
-  entityId?: ToString<EntityID>
+  dialogId?: ToString<EntityID> // This is the entity ID + a prefix indicating the type of dialog
+  accessHash?: string
```

Comment:
If this is required for the job params I'd define a seperate type that extends this type and makes this required.